RecoEgamma/ElectronIdentification/plugins/ElectronMVANtuplizer.cc

```diff
@@ -47,20 +43,12 @@
 // class declaration
 //
 
-// If the analyzer does not use TFileService, please remove
-// the template argument to the base class so the class inherits
-// from  edm::one::EDAnalyzer<>
-// This will improve performance in multithreaded jobs.
-//
-
 class ElectronMVANtuplizer : public edm::one::EDAnalyzer<edm::one::SharedResources>  {
    public:
       explicit ElectronMVANtuplizer(const edm::ParameterSet&);
-      ~ElectronMVANtuplizer() override;
```

Comment:
Even if it is a "one" module, why to remove the destructor?

Reply:
A counter argument, why declare it if the compiler can write the default?

Reply:
I would naively say that otherwise the not overridden dtor of the base class is used instead, and this would fail in deleting specific members of the derived one: isn't it so? Or a virtual method is always overridden by a default one in the derived class?

Reply:
The compiler will aways write a destructor for a class if you do not specifically state you will write the destructor. This is also true for virtual inheritance. I.e. the compiler will write the correct destructor for ElectronMVANtuplizer which will call the destructors for all member data of ElectronMVANtuplizer.

Reply:
Thank you @Dr15Jones ! Comment retracted, then